akka-http-core/src/main/scala/akka/http/Http.scala

```diff
@@ -128,44 +135,44 @@ class HttpExt(config: Config)(implicit system: ActorSystem) extends akka.actor.E
+
     val transportFlow = StreamTcp().outgoingConnection(remoteAddr, localAddress,
       options, effectiveSettings.connectingTimeout, effectiveSettings.idleTimeout)
-    val clientBluePrint = HttpClient.clientBlueprint(remoteAddr, effectiveSettings, log)
-
-    Flow(transportFlow, clientBluePrint)(Keep.left) { implicit b ⇒
-      (tcp, client) ⇒
-        import FlowGraph.Implicits._
 
-        tcp.outlet ~> client.bytesIn
-        client.bytesOut ~> tcp.inlet
-
-        (client.httpRequests, client.httpResponses)
-    }.mapMaterialized { tcpConnFuture ⇒
+    layer.joinMat(transportFlow) { (_, tcpConnFuture) ⇒
       import system.dispatcher
-      tcpConnFuture.map { tcpConn ⇒ OutgoingConnection(tcpConn.localAddress, tcpConn.remoteAddress) }
+      tcpConnFuture map { tcpConn ⇒ OutgoingConnection(tcpConn.localAddress, tcpConn.remoteAddress) }
     }
-
   }
 
   /**
-   * Transforms the given low-level TCP client transport [[Flow]] into a higher-level HTTP client flow.
+   * The type of the client-side HTTP layer as a stand-alone BidiStage
+   * that can be put atop the TCP layer to form an HTTP client.
+   *
+   * {{{
+   *                +------+
+   * HttpRequest  ~>|      |~> ByteString
+   *                | bidi |
+   * HttpResponse <~|      |<~ ByteString
+   *                +------+
+   * }}}
    */
-  def transportToConnectionClientFlow[Mat](transport: Flow[ByteString, ByteString, Mat],
-                                           remoteAddress: InetSocketAddress, // TODO: removed after #16168 is cleared
-                                           settings: Option[ClientConnectionSettings] = None,
-                                           log: LoggingAdapter = system.log): Flow[HttpRequest, HttpResponse, Mat] = {
-    val effectiveSettings = ClientConnectionSettings(settings)
-    val clientBlueprint = HttpClient.clientBlueprint(remoteAddress, effectiveSettings, log)
+  type ClientLayer = BidiFlow[HttpRequest, ByteString, ByteString, HttpResponse, Any]
 
-    Flow(clientBlueprint, transport)(Keep.right) { implicit b ⇒
-      (client, tcp) ⇒
-        import FlowGraph.Implicits._
-        client.bytesOut ~> tcp.inlet
-        tcp.outlet ~> client.bytesIn
+  /**
+   * Constructs a [[ClientLayer]] stage using the configured default [[ClientConnectionSettings]].
+   */
+  def clientLayer(remoteAddress: InetSocketAddress /* TODO: remove after #16168 is cleared */ ): ClientLayer =
+    clientLayer(remoteAddress, ClientConnectionSettings(system))
 
-        (client.httpRequests, client.httpResponses)
-    }
-  }
+  /**
+   * Constructs a [[ClientLayer]] stage using the given [[ClientConnectionSettings]].
+   */
+  def clientLayer(remoteAddress: InetSocketAddress, // TODO: remove after #16168 is cleared
+                  settings: ClientConnectionSettings,
+                  log: LoggingAdapter = system.log): ClientLayer =
+    BidiFlow.wrap(OutgoingConnectionBlueprint(remoteAddress, settings, log))
```

Comment:
No reason. We could easily move the `BidiFlow.wrap` into `OutgoingConnectionBlueprint` if you'd prefer that.